update_cache.py

```diff
@@ -64,10 +64,25 @@ def main():
         sys.exit(1)
 
     # let's grab sha and version info as Tuxsuite has the most up-to-date info
-    # we only need the first entry
-    first_entry = builds[next(iter(builds))]
-    git_sha = first_entry["git_sha"]
-    clang_version = first_entry["tuxmake_metadata"]["compiler"]["version_full"]
+    builds_that_are_missing_metadata = []
+    git_sha = None
+    clang_version = None
+    for entry, build in builds.items():
+        try:
+            git_sha = build["git_sha"]
+            clang_version = build["tuxmake_metadata"]["compiler"][
+                "version_full"]
+        except KeyError:
+            builds_that_are_missing_metadata.append(entry)
+
+    if len(builds_that_are_missing_metadata) > 0:
+        print(
```

Comment:
The string here is going to have a very long first line. It might be better to print something like this:

```
    print(f"""\
Error: These builds in {raw} are missing metadata:
{builds_that_are_missing_metadata}
""")
```

Think about those poor people who are still using 80-column monitors! :-)

Reply:
`raw` is a giant json dump. `builds_that_are_missing_metadata` is just a list of hashes for the builds with malformed metadata.

Here's some example output (trimmed):


```
[4.14 (clang-13)/TuxSuite (defconfigs)]   🐳  docker exec cmd=[sh -e /var/run/act/workflow/4.sh] user= workdir=
| Reading mock.builds.json
| Error: Some of the builds in builds.json are malformed and missing some metadata. Here's a list: ['2YlyQFeMIAc94Pn5vFMTT2hW7Kt']
| Here's the build.json in question:
| {
|     "builds": {
|         "2YlyQFeMIAc94Pn5vFMTT2hW7Kt": {
|             "auto_retry": true,
|             "build_name": "",
|             "build_status": "pass",
|             "callback": null,
|             "client_token": "23427fe5-0d8c-4df1-b2f5-33d349e29027",
...
(snip)
```

Reply:
Oh...ew...Is that useful for debugging? I guess if this is for a webpage it would be... I'll leave it to your judgement. :-)

Reply:
@nathanchance and I spoke about this and in an event of a failure like what we experienced with: https://github.com/ClangBuiltLinux/continuous-integration2/actions/runs/7537720415/job/20517094849 we would prefer if the builds.json was dumped. 

So yes, basically only for web usage.